source/state_representation/README.md

```diff
@@ -372,7 +382,28 @@ Direct multiplication of the `Jacobian` object with another `Eigen::MatrixXd` ha
 `Eigen::MatrixXd` product.
 
 ```cpp
-state_representation::Jacobian jac("myrobot", 3, Eigen::MatrixXd::Random(6, 3));
+state_representation::Jacobian jac("myrobot", 3, "eef", Eigen::MatrixXd::Random(6, 3));
+// alternatively can use the Random static constructor
+state_representation::Jacobian jac = state_representation::Jacobian::Random("myrobot", 3, "eef");
 Eigen::MatrixXd mat = jac.data();
 Eigen::MatrixXd res = jac * Eigen::MatrixXd(3, 4); // equivalent to  jac.data() * Eigen::MatrixXd(3, 4);
 ```
+
+### Changing the Jacobian reference frame
+
+As stated earlier, the `Jacobian` is expressed in a reference frame and can, therefore, use the operations with
+`CartesianPose` to be modified. It relies on the usage of the overloaded `operator*` or the `set_reference_frame`
+function for inplace modifications. It is equivalent to multiply each columns of the `Jacobian` by the `CartesianPose`
+on both the linear and angular part of the matrix. For this operation to be valid, the `CartesianPose` name has to
+match the current `Jacobian` reference frame.
+
+```cpp
+state_representation::Jacobian jac = state_representation::Jacobian::Random("myrobot", 3, "eef", "base_frame");
+state_representation::CartesianPose pose = state_representation::CartesianPose::Random("base_frame", "world");
+// the result is the Jacobian expressed in world
+state_representation::Jacobian jac_in_world = pose * jac;
```

Comment:
I would prefer this multiplication the other way round. Like this it would be `eef` - `base_frame` (from jacobian) times `base_frame` - `world` (pose) and it would be clear that the result is `eef` in `world` because the `base_frame` cancel. At least that's how I like to imagine these transformations. What do you think?

Reply:
using the AinB notation is easiest when writing it in plain text, but really I think of it like bA (with a preceding subscript)

So if `base_frame_in_world` * `eef_in_base_frame` is AinB * CinA, you can reformat that as `bA * aC`. And, the premultiplication of the transform lines up with our operation on vectors too (`M * v`, not `v * M`)

Reply:
Yes exactly the multiplication on the right side would have a different meaning thought. I like to see them as pre and post-fix as:

```
w_J_eef = w_T_base * base_J_eef
base_J_gripper = base_J_eef * eef_T_gripper
```

Is that clearer? Also the second operation is actually not implemented atm

Reply:
But I think your problem is actually the order in which the parameters of the `CartesianState` and `Jacobian` are defined. Remember the second is the `reference_frame` hence:

```
CartesianPose world_T_base = CartesianPose("base", "world")
```

It is designed like this as the reference frame is an optional parameter (default `"world"`) and in c++ optional parameters are always at the end unlike in python. We could change that but this would be a damn refactor and would mean the `reference_frame` become compulsory.

Reply:
I see, that all makes sense. The last answer makes most sense to me yes, but let's not go down this road haha